src/Core/Security/Authentication/Application/UseCase/Login/Login.php

```diff
@@ -0,0 +1,219 @@
+
+final class Login
+{
+    use LoggerTrait;
+
+    /** @var ProviderAuthenticationInterface */
+    private ProviderAuthenticationInterface $provider;
+
+    /**
+     * @param ProviderAuthenticationFactoryInterface $providerFactory
+     * @param SessionInterface $session
+     * @param DataStorageEngineInterface $dataStorageEngine
+     * @param WriteSessionRepositoryInterface $sessionRepository
+     * @param ReadTokenRepositoryInterface $readTokenRepository
+     * @param WriteTokenRepositoryInterface $writeTokenRepository
+     * @param WriteSessionTokenRepositoryInterface $writeSessionTokenRepository
+     * @param AclUpdaterInterface $aclUpdater
+     */
+    public function __construct(
+        private ProviderAuthenticationFactoryInterface $providerFactory,
+        private SessionInterface $session,
+        private DataStorageEngineInterface $dataStorageEngine,
+        private WriteSessionRepositoryInterface $sessionRepository,
+        private ReadTokenRepositoryInterface $readTokenRepository,
+        private WriteTokenRepositoryInterface $writeTokenRepository,
+        private WriteSessionTokenRepositoryInterface $writeSessionTokenRepository,
+        private AclUpdaterInterface $aclUpdater
+    )
+    {
+    }
+
+    /**
+     * @param LoginRequest $loginRequest
+     * @param PresenterInterface $presenter
+     * @return void
+     * @throws AuthenticationException
+     */
+    public function __invoke(LoginRequest $loginRequest, PresenterInterface $presenter): void
+    {
+        try {
+            $this->provider = $this->providerFactory->create($loginRequest->providerName);
+
+            $this->provider->authenticateOrFail($loginRequest);
+            $user = $this->provider->findUserOrFail();
+
+            if ($loginRequest->providerName === Provider::LOCAL && !$user->isAllowedToReachWeb()) {
+                throw LegacyAuthenticationException::notAllowedToReachWebApplication();
+            }
+
+            if ($this->provider->isAutoImportEnabled()) {
+                $this->provider->importUserToDatabase();
+            }
+
+            $this->updateACL($user);
+
+            // Start a new session
+            if ($this->sessionRepository->start($this->provider->getLegacySession())) {
+                if ($this->readTokenRepository->hasAuthenticationTokensByToken($this->session->getId()) === false) {
+                    $this->createAuthenticationTokens(
+                        $this->session->getId(),
+                        $user,
+                        $this->provider->getProviderToken(),
+                        $this->provider->getProviderRefreshToken(),
+                        $loginRequest->clientIp
+                    );
+                }
+            }
+        } catch (PasswordExpiredException $e) {
+            $response = new PasswordExpiredResponse($e->getMessage());
+            $response->setBody([
+                'password_is_expired' => true,
+            ]);
```

Comment:
```suggestion
            $response->passwordIsExpired = true;
```
then your presenter should handle the presentation of your array